From a6334c0024daa6a87b36e2309aaa50f2e04d49ce Mon Sep 17 00:00:00 2001
From: sunchendong <sunchendong@xfusion.com>
Date: Wed, 24 Dec 2025 16:49:05 +0800
Subject: [PATCH] [xfusion]vLLM-ascend:fix Suffix Decoding

Offering:AI Solution 25
CVE:
Reference:
Type:requirement
DTS/US:US20251111238969
reason:vLLM-ascend:fix Suffix Decoding

Signed-off-by: sunchendong <sunchendong@xfusion.com>
---
 vllm_ascend/utils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vllm_ascend/utils.py b/vllm_ascend/utils.py
index ef74c35..86a4c63 100644
--- a/vllm_ascend/utils.py
+++ b/vllm_ascend/utils.py
@@ -475,7 +475,7 @@ def update_aclgraph_sizes(vllm_config: VllmConfig) -> None:
 
     # Calculate maximum supported batch sizes considering model architecture
     resources_per_graph = num_hidden_layers + 1
-    if vllm_config.speculative_config is not None:
+    if vllm_config.speculative_config is not None and vllm_config.speculative_config.draft_model_config is not None:
         draft_model_hf_config = vllm_config.speculative_config.draft_model_config.hf_config
         resources_per_graph += draft_model_hf_config.num_hidden_layers + 1
 
-- 
2.43.0

